frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -997,6 +1000,14 @@ private void removeContainerRequest(TaskStatus taskStatus) {
     } catch (Exception e) {
       LOGGER.logError(e, "%s: Failed to removeContainerRequest", taskLocator);
     }
+    if(request.getNodes() != null && request.getNodes().size() != 0) {
+      try {
+        ResourceDescriptor resourceDescriptor = ResourceDescriptor.fromResource(request.getCapability());
+        gpuAllocationManager.releaseRequestingResource(resourceDescriptor, request.getNodes());
+      } catch (Exception e) {
+        LOGGER.logError(e, "%s: Failed to ResourceDescriptor.fromResource", taskLocator);
```

Comment:
>ResourceDescriptor.fromResource [](start = 42, length = 31)

can still consider it as a whole, so:
LOGGER.logError(e, "%s: Failed to releaseRequestingResource